core/src/main/java/io/questdb/griffin/engine/groupby/vect/AvgDoubleVectorAggregateFunction.java

```diff
@@ -55,16 +58,20 @@ public AvgDoubleVectorAggregateFunction(int keyKind, int columnIndex, int worker
             distinctFunc = Rosti::keyedIntDistinct;
             keyValueFunc = Rosti::keyedIntSumDouble;
         }
-
+        counts = Unsafe.malloc(workerCount * 8L, MemoryTag.NATIVE_DEFAULT);
```

Comment:
We should use `(long) Misc.CACHE_LINE_SIZE` instead of `8L` here and in other `Avg*` functions. That's to prevent false sharing.

A side note. I've noticed weird looking code around `Misc.CACHE_LINE_SIZE` in `SumDoubleVectorAggregateFunction` and other functions. They seem to overallocate arrays: e.g. `new long[workerCount * Misc.CACHE_LINE_SIZE]` while it's enough to allocate `new long[workerCount * (Misc.CACHE_LINE_SIZE / Long.BYTES)]`. Am I missing anything?

Reply:
Will be addressed in a separate PR.